orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -328,6 +328,55 @@ void addObjectMethodsForTraining(py::module& m) {
         }
         return rmap;
       })
+      .def("get_model_state", [](PyTrainingSession* sess, bool include_mixed_precision_weights) {
+        std::unordered_map<std::string, NameMLValMap> model_state_tensors;
+        ORT_THROW_IF_ERROR(static_cast<TrainingSession*>(sess->GetSessionHandle())->GetModelState(model_state_tensors, include_mixed_precision_weights));
+        auto& data_transfer_manager = sess->GetSessionHandle()->GetDataTransferManager();
+        //convert to numpy array
+        std::unordered_map<std::string, std::unordered_map<std::string, py::object>> py_model_state;
+        for (const auto& fp_or_mp_tensors: model_state_tensors) {
+          std::unordered_map<std::string, py::object> weight_tensor_map;
+          for (const auto& weight: fp_or_mp_tensors.second) {
+            if (weight.second.IsTensor()) {
+              py::object obj;
+              const Tensor& rtensor = weight.second.Get<Tensor>();
+              GetPyObjFromTensor(rtensor, obj, &data_transfer_manager);
+              weight_tensor_map.insert({weight.first, obj});
+            } else {
+              throw std::runtime_error("Non tensor type in session state tensors is not expected.");
+            }
+          }
+          py_model_state.insert({fp_or_mp_tensors.first, weight_tensor_map});
+        }
+        return py_model_state;
+      })
+      .def("get_optimizer_state", [](PyTrainingSession* sess) {
+        std::unordered_map<std::string, NameMLValMap> opt_state_tensors;
+        ORT_THROW_IF_ERROR(static_cast<TrainingSession*>(sess->GetSessionHandle())->GetOptimizerState(opt_state_tensors));
+        auto& data_transfer_manager = sess->GetSessionHandle()->GetDataTransferManager();
+        //convert to numpy array
+        std::unordered_map<std::string, std::unordered_map<std::string, py::object>> py_optimizer_tensors;
```

Comment:
the `get_model_state` and `get_optimizer_state` have some repeated code, can we move it to a function call?

Reply:
Moved to a new function `ConvertORTTensorMapToNumpy`